src/bootstrap/Cargo.toml

```diff
@@ -36,7 +36,7 @@ test = false
 # Most of the time updating these dependencies requires modifications
 # to the bootstrap codebase; otherwise, some targets will fail. That's
 # why these dependencies are explicitly pinned.
-cc = "=1.0.73"
+cc = "=1.0.90"
```

Comment:
As this comment indicates, I'm quite certain that some of the targets will fail during the merge CI.

Reply:
I'm happy to work on fixes, but I'll probably need some guidance on how to test locally. I have Linux, Mac, and Windows machines at my disposal, and I'm motivated to get the cc crate version upgraded, so we don't have to keep maintaining local patches.

Reply:
Debugging this isn't straightforward since we're testing various targets in CI. I guess as a start you can try to replicate `dist-apple-various` (which failed couple times on cc bump, ref: https://github.com/rust-lang/rust/pull/119654#issuecomment-1885930572) locally . Once it's fixed, we can queue this PR with rollup=never. If merge is successful, that's great; if not, we can continue with debugging the new error.

Reply:
@onur-ozkan cc-rs was not running tests on aarch64-apple-darwin before 1.0.86, and it now is, and fixed several critical issues with building on Apple in the versions immediately before that. The cc version was already bumped for the workspace, so I am struggling to believe that bootstrap is really that special of a snowflake here, given that cc-rs is used to complete the `rustc_llvm` build.

Reply:
iirc I was getting errors related to deprecated/changed env variables. We can r+ and see I guess?

Reply:
I'm having trouble replicating the dist-apple-various error, but I at least seem to be getting a vaguely related error. It looks like the error is related to cross-contamination between MacOS and iOS builds and object files. I am getting a linker error (ld: building for 'macOS', but linking in object file (blah) built for 'iOS')

But roughly the same thing happens whether I am using cc 1.0.73 or 1.0.90, so I'm hesitant to conclude too much.